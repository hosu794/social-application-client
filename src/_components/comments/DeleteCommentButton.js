import React from "react";

import { commentActions } from "../../_actions";

import { useDispatch } from "react-redux";

function DeleteCommentButton({ handleDeleteComment }) {
  return (
    <button onClick={handleDeleteComment} className="button is-danger">
      Delete
    </button>
  );
}

export default DeleteCommentButton;
