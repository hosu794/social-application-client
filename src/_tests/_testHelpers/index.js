export * from "./mockStore";
export * from "./mockStory";
export * from "./mockComment";
export * from "./mockIdMatchParams";
