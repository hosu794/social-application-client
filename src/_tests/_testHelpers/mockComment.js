export let mockComment = {
  id: 1,
  body: "Some body",
  createdBy: {
    id: 12,
    name: "Joe Doe",
    username: "exampleusername",
    downloadAvatar: "dsdasdasdasdasdadas.png",
  },
};
