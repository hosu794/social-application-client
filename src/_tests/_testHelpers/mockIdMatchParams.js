export let mockIdMatchParams = {
  params: {
    id: 12,
  },
};
