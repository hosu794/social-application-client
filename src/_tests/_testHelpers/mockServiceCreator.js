export const mockServiceCreator = (body, succeeds = true) => () =>
  new Promise((resolve, reject) => {
    setTimeout(() => (succeeds ? resolve(body) : reject(body)), 10);
  });
