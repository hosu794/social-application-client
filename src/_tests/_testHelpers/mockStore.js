import React from "react";

import configureStore from "redux-mock-store";
import { useDispatch } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import Comment from "../../_components/comments/Comment";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import Comments from "../../_components/comments/Comments";
import CreateComment from "../../_components/comments/CreateComment";

const mockStore = configureStore([]);

export let store = mockStore({
  alert: {
    type: "alert-success",
  },
  comment: {
    loaded: true,
    currentComment: {
      id: 1,
      body: "Some body",
      createdBy: {
        id: 12,
        name: "Joe Doe",
        username: "exampleusername",
        downloadAvatar: "dsdasdasdasdasdadas.png",
      },
    },
    content: [
      {
        id: 1,
        body: "Some body",
        createdBy: {
          id: 12,
          name: "Joe Doe",
          username: "exampleusername",
          downloadAvatar: "dsdasdasdasdasdadas.png",
        },
      },
    ],
  },
  user: {
    isUverLovedStory: true,
    user: {
      id: 12,
      name: "Joe Doe",
      username: "exampleusername",
      downloadAvatar: "dsdasdasdasdasdadas.png",
    },
  },
  authentication: {
    loggedIn: true,
  },
  stories: {
    content: [],
    currentStory: {
      id: 1,
      title: "Story Title",
      description: "Story description",
      totalLoves: 12,
      body: "<p>Simple Story</p>",
      createdBy: {
        id: 12,
        name: "Joe Doe",
        username: "exampleusername",
        downloadAvatar: "dsdasdasdasdasdadas.png",
      },
      topic: {
        id: 12,
        title: "Topic Example Title",
        description: "Description Example description",
        createdAt: "Dummy date",
        createdBy: {
          id: 12,
          name: "Joe Doe",
          username: "exampleusername",
          downloadAvatar: "dsdasdasdasdasdadas.png",
        },
      },
    },
  },
});
