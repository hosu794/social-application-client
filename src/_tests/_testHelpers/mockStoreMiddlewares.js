import thunk from "redux-thunk";

import configureMockStore from "redux-mock-store";

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

export let storeMiddlewares = mockStore({});
