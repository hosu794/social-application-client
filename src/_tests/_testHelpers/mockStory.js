export const mockStory = {
  id: 1,
  title: "Story Title",
  description: "Story description",
  totalLoves: 12,
  body: "<p>Simple Story</p>",
  createdBy: {
    id: 12,
    name: "Joe Doe",
    username: "exampleusername",
    downloadAvatar: "dsdasdasdasdasdadas.png",
  },
  topic: {
    id: 12,
    title: "Topic Example Title",
    description: "Description Example description",
    createdAt: "Dummy date",
    createdBy: {
      id: 12,
      name: "Joe Doe",
      username: "exampleusername",
      downloadAvatar: "dsdasdasdasdasdadas.png",
    },
  },
  authentication: {
    loggedIn: true,
  },
};
