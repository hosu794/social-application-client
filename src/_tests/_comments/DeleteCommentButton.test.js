import React from "react";

import { Provider } from "react-redux";

import { mount } from "enzyme";

import { mockStory, mockComment, store } from "../_testHelpers";
import DeleteCommentButton from "../../_components/comments/DeleteCommentButton";
import { MemoryRouter } from "react-router-dom";

const mockCallBack = jest.fn();

describe("Tests for the DeleteCommentButton Component", () => {
  let TestingComponent = () => (
    <Provider store={store}>
      <MemoryRouter>
        <DeleteCommentButton handleDeleteComment={mockCallBack} />
      </MemoryRouter>
    </Provider>
  );

  let wrapper = mount(<TestingComponent />);

  test("should component render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(DeleteCommentButton).toBeDefined();
  });

  test("should call function on button click", () => {
    wrapper.find("button").props().onClick();

    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
