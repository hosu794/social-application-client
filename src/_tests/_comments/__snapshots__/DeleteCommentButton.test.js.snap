// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the DeleteCommentButton Component should component render correctly 1`] = `ReactWrapper {}`;
